pkg/river/encoding/array_test.go

```diff
@@ -0,0 +1,39 @@
+package encoding
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/grafana/agent/pkg/river/internal/value"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSimpleArray(t *testing.T) {
+	intArr := make([]int, 10)
+	for i := 0; i < 10; i++ {
+		intArr[i] = i
+	}
+	reqString := `{"type":"array","value":[{"type":"number","value":0},{"type":"number","value":1},{"type":"number","value":2},{"type":"number","value":3},{"type":"number","value":4},{"type":"number","value":5},{"type":"number","value":6},{"type":"number","value":7},{"type":"number","value":8},{"type":"number","value":9}]}`
```

Comment:
Can we split this across multiple lines to make it easier to read? JSONEq ignores formatting of the JSON strings. Ditto with the other tests in this file and other files.

Reply:
It's a lot more verbose to format it out adds 150 lines but definitely can.